web/src/components/vis/snippets/BoxplotBox.vue

```diff
@@ -0,0 +1,59 @@
+<script lang="ts">
+import Vue, { PropType } from 'vue';
+
+export interface BoxData {
+  whiskers: { d: string }[];
+  lines: { x1: number; x2: number }[];
+  circles: { cx: number }[];
+}
+
+export default Vue.extend({
+  props: {
+    data: {
+      type: Object as PropType<BoxData>,
+      required: true,
+    },
+  },
+});
```

Comment:
I'm not really up to date on the composition API pros/cons, but would it be better to use it for all new components, or are we only using it where there's a very convincing reason to? This component is trivially small so it's not really important here, I'm just curious about our general philosophy.

Reply:
So far, I've been following the guidance that it's only worth using if you need mixin-like behavior a-la `useAxisPlot`, otherwise best to stick to object API.

Reply:
@subdavis, is that guidance specific to Viime and our cold-storage prep? In general I would think abandoning object API would be a long-term goal.

Reply:
> In general I would think abandoning object API would be a long-term goal.

That guidance is general.  Object API is here to stay.  It reads better and it's less flexible than composition which leads to higher quality code in the average case.  

There's a little bit of fighting right now between them because Vue 2's object API passes props that use a reactivity implementation different than how composition works, but in Vue 3 it doesn't matter which style you use, refs will be refs, so you can switch between them freely.